ocean_provider/validation/RBAC.py

```diff
@@ -0,0 +1,146 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import json
+from typing import Optional
+
+from ocean_lib.common.agreements.service_types import ServiceTypesIndices
+from ocean_lib.web3_internal.transactions import sign_hash
+
+from ocean_provider.exceptions import RequestNotFound
+from ocean_provider.utils.basics import get_provider_wallet
+from ocean_provider.utils.util import msg_hash
+
+
+class RBACValidator:
+    def __init__(
+        self,
+        request_name=None,
+        request=None,
+        payload: Optional[dict] = None,
+        assets: Optional[list] = None,
+        algorithms: Optional[list] = None,
+    ):
+        self.request = request
+        self.payload = payload if payload else dict()
+        if not request:
+            raise RequestNotFound("Request not found.")
+        action_mapping = self.get_action_mapping()
+        self.action = action_mapping[request_name]
+        self.credentials = {
+            "type": "address",
+            "address": get_provider_wallet().address,
+        }
+        self.component = "provider"
+        self.provider_address = get_provider_wallet().address
+        self.assets = assets if assets else []
+        self.algorithms = algorithms if algorithms else []
+        if request_name in ["ComputeRequest", "ComputeStartRequest"]:
+            self.additional_inputs = self.payload["additionalInputs"]
+
+    @staticmethod
+    def get_action_mapping():
+        action_mapping = {
+            "EncryptRequest": "encryptUrl",
+            "InitializeRequest": "initialize",
+            "DownloadRequest": "access",
+            "ComputeRequest": "compute",
+            "ComputeStartRequest": "compute",
+        }
+        return action_mapping
+
+    def fails(self):
+        return False
+
+    def get_compute_dict(self):
+        return {
+            "dids": [
+                {
+                    "did": asset.did,
+                    "serviceId": ServiceTypesIndices.DEFAULT_COMPUTING_INDEX,
+                }
+                for asset in self.assets
+            ],
+            "algos": [
+                {
+                    "did": algorithm.did,
+                    "serviceId": ServiceTypesIndices.DEFAULT_COMPUTING_INDEX,
+                }
+                for algorithm in self.algorithms
+            ],
+            "additionalDids": [
+                {
+                    "did": additional_input["documentId"],
+                    "serviceId": additional_input["serviceId"],
+                }
+                for additional_input in self.additional_inputs
+            ],
+        }
+
+    def get_dids(self):
+        return {
```

Comment:
I don't understand why these are dictionaries with a single key? And used by accessing that key?
I mean you have `get_dids(self): return {"dids": { "foo": "bar"} }` and then you use it as `get_dids()["dids"]`
Why not return {"foo": "bar"}  and use it directly?

The same goes for the compute dict, which again I did not understand why is used in the exact same way.

Also, why not paremeterise the serviceId value? You could reuse like this: get_dids(ServiceTypesIndices.DEFAULT_ACCESS_INDEX) or with the other index. i.e. a parameter called serviceTypeIndex